web/src/screens/private/Groups/EditGroup.js

```diff
@@ -3,39 +3,328 @@ import PropTypes from 'prop-types';
 import { compose } from 'recompose';
 import { graphql } from 'react-apollo';
 import { connect } from 'react-redux';
+import { NavLink } from 'react-router-dom';
+import _ from 'lodash';
 
+// material ui
 import { withStyles } from 'material-ui/styles';
 import { CircularProgress } from 'material-ui/Progress';
+import Typography from 'material-ui/Typography';
+import TextField from 'material-ui/TextField';
+import { FormControl } from 'material-ui/Form';
+import Button from 'material-ui/Button';
+import Paper from 'material-ui/Paper';
 
-import CURRENT_USER_QUERY from '../../../graphql/current-user.query';
+// gql
+import CURRENT_ORG_QUERY from '../../../graphql/current-org.query';
+import CREATE_GROUP_MUTATION from '../../../graphql/create-group.mutation';
+import UPDATE_GROUP_MUTATION from '../../../graphql/update-group.mutation';
 
-import styles from './EditGroup.styles';
+// components
+import Tag from '../../../components/Selects/Tag';
 
-const EditGroup = ({ classes, loading }) => {
-  if (loading) {
-    return <CircularProgress className={classes.progress} size={50} />;
+// constants
+import { TAG_TYPE_CAPABILITY, TAG_TYPE_ORG_STRUCTURE } from '../../../constants';
+
+import styles from '../../../styles/AppStyle';
+
+class EditGroup extends React.Component {
+  state = {
+    id: 0,
+    name: '',
+    capabilities: '',
+    regions: '',
+    users: '',
+  };
+
+  componentDidMount() {
+    if (this.props.loading === false && this.props.match.params.id !== undefined) {
+      this.setInitialState(this.props);
+    }
   }
 
-  return <div className={classes.root}>Add Group</div>;
-};
+  componentWillReceiveProps(nextProps) {
+    if (
+      nextProps.loading === false &&
+      this.props.match.params.id !== undefined &&
+      this.state.id === 0
+    ) {
+      this.setInitialState(nextProps);
+    }
+  }
+
+  setInitialState(props) {
+    const group = props.user.organisation.groups.find(
+      g => g.id === parseInt(props.match.params.id, 10),
+    );
+    if (group !== undefined) {
+      let regions = '';
```

Comment:
do we have a plan on how to handle the different name we might want to call this? org_structure vs regions vs zone vs units etc etc?

Reply:
yeah I thought the whole point of this was that we didn't need to hard code these kinds of things :(
Don't want to block progress at this stage but think about it some more if you can.

Reply:
I'm getting a ton of pressure to get this functionality in, I can move it to config in the future but under time pressure at the moment to get these done.